lib/generators/rspec/scaffold/templates/api_controller_spec.rb

```diff
@@ -0,0 +1,165 @@
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new <%= class_name %>" do
+        expect {
+<% if RUBY_VERSION < '1.9.3' -%>
+          post :create, {:<%= ns_file_name %> => valid_attributes}, valid_session
+<% else -%>
+          post :create, params: {<%= ns_file_name %>: valid_attributes}, session: valid_session
+<% end -%>
+        }.to change(<%= class_name %>, :count).by(1)
+      end
+
+      it "renders a JSON response with the new <%= ns_file_name %>" do
+<% if RUBY_VERSION < '1.9.3' -%>
+        post :create, {:<%= ns_file_name %> => valid_attributes}, valid_session
+<% else %>
+        post :create, params: {<%= ns_file_name %>: valid_attributes}, session: valid_session
+<% end -%>
+        expect(response).to have_http_status(:created)
+        expect(response.content_type).to eq('application/json')
+        expect(response.location).to eq(<%= ns_file_name %>_url(<%= class_name %>.last))
+      end
+    end
+
+    context "with invalid params" do
+      it "renders a JSON response with errors for the new <%= ns_file_name %>" do
+<% if RUBY_VERSION < '1.9.3' -%>
+        post :create, {:<%= ns_file_name %> => invalid_attributes}, valid_session
+<% else %>
+        post :create, params: {<%= ns_file_name %>: invalid_attributes}, session: valid_session
+<% end -%>
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response.content_type).to eq('application/json')
+      end
+    end
+  end
+
+  describe "PUT #update" do
+    context "with valid params" do
+      let(:new_attributes) {
+        skip("Add a hash of attributes valid for your model")
+      }
+
+      it "updates the requested <%= ns_file_name %>" do
+        <%= file_name %> = <%= class_name %>.create! valid_attributes
+<% if RUBY_VERSION < '1.9.3' -%>
+        put :update, {:id => <%= file_name %>.to_param, :<%= ns_file_name %> => new_attributes}, valid_session
+<% else -%>
+        put :update, params: {id: <%= file_name %>.to_param, <%= ns_file_name %>: new_attributes}, session: valid_session
+<% end -%>
+        <%= file_name %>.reload
+        skip("Add assertions for updated state")
+      end
+
+      it "renders a JSON response with the <%= ns_file_name %>" do
+        <%= file_name %> = <%= class_name %>.create! valid_attributes
+<% if RUBY_VERSION < '1.9.3' -%>
+        put :update, {:id => <%= file_name %>.to_param, :<%= ns_file_name %> => valid_attributes}, valid_session
+<% else %>
+        put :update, params: {id: <%= file_name %>.to_param, <%= ns_file_name %>: valid_attributes}, session: valid_session
+<% end -%>
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq('application/json')
+      end
+    end
+
+    context "with invalid params" do
+      it "renders a JSON response with errors for the <%= ns_file_name %>" do
+<% if RUBY_VERSION < '1.9.3' -%>
+        <%= file_name %> = <%= class_name %>.create! valid_attributes
```

Comment:
@k3rni 

you missed moving this file name assignment up to outside of the 1.9.3 conditional, so this does not work on ruby interpreters greater than or equal to 1.9.3, can you add a spec for this case and make it pass? Thanks